hydromt/components/table.py

```diff
@@ -0,0 +1,120 @@
+"""Table component."""
+
+import glob
+import os
+from os.path import basename, dirname, join
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+    cast,
+)
+
+import pandas as pd
+
+from hydromt._typing.type_def import PandasLike
+from hydromt.components.base import ModelComponent
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+__all__ = ["TableComponent"]
+
+
+class TableComponent(ModelComponent):
+    """Table Component."""
+
+    def __init__(
+        self,
+        model: "Model",
+    ):
+        """Initialize a TableComponent.
+
+        Parameters
+        ----------
+        model: Model
+            HydroMT model instance
+        """
+        self._data: Optional[Dict[str, PandasLike]] = None
+        super().__init__(model=model)
+
+    @property
+    def data(self) -> Dict[str, PandasLike]:
+        """Model tables."""
+        if self._data is None:
+            self._initialize_tables()
+        if self._data is None:
+            raise RuntimeError("Could not load data for table component")
```

Comment:
Is this check really necessary?

Reply:
not functionally, but it helps the type checkers

Reply:
You can also `assert self._data is not None`

Reply:
And it is also never true, because `_initialize_tables()` will always set `self._data` to an empty dict I believe

Reply:
Indeed though my type checker doesn't seem to pick up on that fact

Reply:
I meant the same as @deltamarnix that the check should not be necessary. Though then I don't understand your answer @savente93. Is there an actual bug or error if you remove that check? Seems a bit worrisome...